master/buildbot/worker/protocols/pb.py

```diff
@@ -191,47 +191,58 @@ def remotePrint(self, message):
+                if info is None:
+                    defer.returnValue(None)
         except _NoSuchMethod:
-            info = {}
-
-            # Probably this is deprecated buildslave.
-            log.msg("Worker.getWorkerInfo is unavailable - falling back to "
-                    "deprecated buildslave API")
+            # this is a really old worker
+            log.msg("Worker.getSlaveInfo is unavailable - ignoring")
 
-            try:
-                with _wrapRemoteException():
-                    info = yield self.mind.callRemote('getSlaveInfo')
-            except _NoSuchMethod:
-                log.msg("Worker.getSlaveInfo is unavailable - ignoring")
+        # newer workers send all info in one command
+        if "slave_commands" in info:
+            assert "worker_commands" not in info
+            info["worker_commands"] = info.pop("slave_commands")
+            defer.returnValue(info)
 
-            # newer workers send all info in one command
-            if "slave_commands" in info:
-                assert "worker_commands" not in info
-                info["worker_commands"] = info.pop("slave_commands")
-                defer.returnValue(info)
+        # Old version buildslave - need to retrieve list of supported
+        # commands and version using separate requests.
+        try:
+            with _wrapRemoteException():
+                info["worker_commands"] = yield self.mind.callRemote(
+                    'getCommands')
+        except _NoSuchMethod:
+            log.msg("Worker.getCommands is unavailable - ignoring")
 
-            # Old version buildslave - need to retrieve list of supported
-            # commands and version using separate requests.
-            try:
-                with _wrapRemoteException():
-                    info["worker_commands"] = yield self.mind.callRemote(
-                        'getCommands')
-            except _NoSuchMethod:
-                log.msg("Worker.getCommands is unavailable - ignoring")
+        try:
+            with _wrapRemoteException():
+                info["version"] = yield self.mind.callRemote('getVersion')
+        except _NoSuchMethod:
+            log.msg("Worker.getVersion is unavailable - ignoring")
 
-            try:
-                with _wrapRemoteException():
-                    info["version"] = yield self.mind.callRemote('getVersion')
-            except _NoSuchMethod:
-                log.msg("Worker.getVersion is unavailable - ignoring")
+        defer.returnValue(info)
 
+    @defer.inlineCallbacks
+    def remoteGetWorkerInfo(self):
+        # we need to first try the legacy command in order
+        # to avoid legacy workers to output a stack trace in the logs
+        info = yield self.remoteGetWorkerInfoLegacy()
+        if info is not None:
+            # instead of logging, we mark the version as deprecated in the workerinfo
+            info["version"] = "deprecated:" + info.get("version", "unknown version")
```

Comment:
Wouldn't change of version for older workers break something?
I would expect that some users may actually rely on the fact that slave version is the same as version of installed `buildbot-slave` package.

And what is the benefit of having changed version string?
If we want to warn users in the UI I would expect to have added check and custom renderer on the UI side.